~/workspace$ poetry run mypy .
apps/users/managers.py:13: error: Signature of "create_user" incompatible with supertype "django.contrib.auth.models.UserManager"  [override]
apps/users/managers.py:13: note:      Superclass:
apps/users/managers.py:13: note:          def create_user(self, username: str, email: str | None = ..., password: str | None = ..., **extra_fields: Any) -> Any
apps/users/managers.py:13: note:      Subclass:
apps/users/managers.py:13: note:          def create_user(self, email: Any, password: Any = ..., **extra_fields: Any) -> Any
apps/users/managers.py:38: error: Signature of "create_superuser" incompatible with supertype "django.contrib.auth.models.UserManager"  [override]
apps/users/managers.py:38: note:      Superclass:
apps/users/managers.py:38: note:          def create_superuser(self, username: str, email: str | None = ..., password: str | None = ..., **extra_fields: Any) -> Any
apps/users/managers.py:38: note:      Subclass:
apps/users/managers.py:38: note:          def create_superuser(self, email: Any, password: Any = ..., **extra_fields: Any) -> Any
apps/core/services.py:158: error: Returning Any from function declared to return "Model | None"  [no-any-return]
apps/core/services.py:182: error: "Model" has no attribute "id"  [attr-defined]
apps/core/services.py:227: error: "Model" has no attribute "id"  [attr-defined]
apps/core/services.py:268: error: "Model" has no attribute "id"  [attr-defined]
apps/core/services.py:271: error: "Model" has no attribute "id"  [attr-defined]
apps/core/models.py:15: error: Need type annotation for "created_at"  [var-annotated]
apps/core/models.py:18: error: Need type annotation for "updated_at"  [var-annotated]
apps/core/models.py:21: error: Need type annotation for "is_active"  [var-annotated]
apps/core/models.py:52: error: Need type annotation for "owner"  [var-annotated]
apps/core/mixins.py:217: error: "BaseCache" has no attribute "delete_pattern"  [attr-defined]
apps/users/models.py:35: error: Need type annotation for "phone"  [var-annotated]
apps/users/models.py:39: error: Need type annotation for "country"  [var-annotated]
apps/users/models.py:43: error: Need type annotation for "is_email_verified"  [var-annotated]
apps/users/models.py:47: error: Need type annotation for "verification_token"  [var-annotated]
apps/users/models.py:55: error: Need type annotation for "token_created_at"  [var-annotated]
apps/mailings/models.py:23: error: Need type annotation for "email"  [var-annotated]
apps/mailings/models.py:25: error: Need type annotation for "full_name"  [var-annotated]
apps/mailings/models.py:27: error: Need type annotation for "comment"  [var-annotated]
apps/mailings/models.py:54: error: Need type annotation for "subject"  [var-annotated]
apps/mailings/models.py:56: error: Need type annotation for "body"  [var-annotated]
apps/mailings/models.py:96: error: Need type annotation for "start_datetime"  [var-annotated]
apps/mailings/models.py:98: error: Need type annotation for "end_datetime"  [var-annotated]
apps/mailings/models.py:100: error: Need type annotation for "status"  [var-annotated]
apps/mailings/models.py:108: error: Need type annotation for "message"  [var-annotated]
apps/mailings/models.py:116: error: Need type annotation for "recipients"  [var-annotated]
apps/mailings/models.py:120: error: Need type annotation for "successfully_sent"  [var-annotated]
apps/mailings/models.py:128: error: Need type annotation for "current_run"  [var-annotated]
apps/mailings/models.py:144: error: "Mailing" has no attribute "get_status_display"  [attr-defined]
apps/mailings/models.py:179: error: Need type annotation for "mailing"  [var-annotated]
apps/mailings/models.py:187: error: Need type annotation for "recipient"  [var-annotated]
apps/mailings/models.py:197: error: Need type annotation for "run_number"  [var-annotated]
apps/mailings/models.py:203: error: Need type annotation for "trigger_type"  [var-annotated]
apps/mailings/models.py:211: error: Need type annotation for "status"  [var-annotated]
apps/mailings/models.py:215: error: Need type annotation for "server_response"  [var-annotated]
apps/mailings/models.py:219: error: Need type annotation for "attempted_at"  [var-annotated]
apps/mailings/models.py:229: error: "Attempt" has no attribute "get_status_display"  [attr-defined]
apps/mailings/signals.py:57: error: "Mailing" has no attribute "owner_id"  [attr-defined]
apps/mailings/signals.py:58: error: "Mailing" has no attribute "owner_id"  [attr-defined]
apps/mailings/signals.py:87: error: "Recipient" has no attribute "owner_id"  [attr-defined]
apps/mailings/signals.py:88: error: "Recipient" has no attribute "owner_id"  [attr-defined]
apps/mailings/services.py:30: error: Incompatible return value type (got "Model | None", expected "Recipient | None")  [return-value]
apps/mailings/services.py:37: error: Incompatible return value type (got "Model | None", expected "Recipient | None")  [return-value]
apps/mailings/services.py:62: error: Incompatible return value type (got "Model | None", expected "Message | None")  [return-value]
apps/mailings/services.py:69: error: Incompatible return value type (got "Model | None", expected "Message | None")  [return-value]
apps/mailings/services.py:94: error: Incompatible return value type (got "Model | None", expected "Mailing | None")  [return-value]
apps/mailings/services.py:101: error: Incompatible return value type (got "Model | None", expected "Mailing | None")  [return-value]
apps/mailings/services.py:137: error: "Model" has no attribute "recipients"  [attr-defined]
apps/mailings/services.py:138: error: "Model" has no attribute "id"  [attr-defined]
apps/mailings/services.py:167: error: "Model" has no attribute "recipients"  [attr-defined]
apps/mailings/services.py:168: error: "Model" has no attribute "id"  [attr-defined]
apps/mailings/services.py:200: error: Incompatible return value type (got "Model | None", expected "Attempt | None")  [return-value]
apps/mailings/forms.py:71: error: "Field" has no attribute "queryset"  [attr-defined]
apps/mailings/forms.py:72: error: "Field" has no attribute "queryset"  [attr-defined]
apps/mailings/forms.py:75: error: "Field" has no attribute "input_formats"  [attr-defined]
apps/mailings/forms.py:76: error: "Field" has no attribute "input_formats"  [attr-defined]
apps/mailings/apps.py:124: error: Skipping analyzing "apscheduler.schedulers.background": module is installed, but missing library stubs or py.typed marker  [import-untyped]
apps/mailings/apps.py:124: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
apps/mailings/apps.py:125: error: Skipping analyzing "django_apscheduler.jobstores": module is installed, but missing library stubs or py.typed marker  [import-untyped]
apps/mailings/apps.py:142: error: Module "django.utils.timezone" does not explicitly export attribute "timedelta"  [attr-defined]
apps/users/forms.py:56: error: "AbstractBaseUser" has no attribute "is_email_verified"  [attr-defined]
apps/users/forms.py:66: error: "AbstractBaseUser" has no attribute "token_created_at"  [attr-defined]
apps/users/views.py:41: error: Incompatible types in assignment (expression has type "_StrPromise", base class "LogoutView" defined the type as "str | None")  [assignment]
apps/users/views.py:61: error: Argument 1 to "build_absolute_uri" of "HttpRequest" has incompatible type "_StrPromise"; expected "str | None"  [arg-type]
apps/mailings/views.py:88: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:88: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:113: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:113: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:120: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:120: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:125: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:125: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:151: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:151: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:196: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:196: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:242: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:242: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:249: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:249: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:254: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:254: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:282: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:282: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:309: error: Item "AbstractBaseUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]
apps/mailings/views.py:309: error: Item "AnonymousUser" of "AbstractBaseUser | AnonymousUser" has no attribute "is_manager"  [union-attr]